src/audio/audio_context.rs

```diff
@@ -0,0 +1,37 @@
+//! Provides a context for loading a file.
+
+use super::source::Source;
+use assets::*;
+
+use std::sync::Arc;
+
+/// A context for loading audio files
+pub struct AudioContext {
+    inner: SimpleContext<Source, Vec<u8>, NoError, fn(Vec<u8>) -> Result<Source, NoError>>,
```

Comment:
Oh yeah sorry, I'm pretty inaccurate about that. The problem with a runtime function pointer is that inlining isn't possible. And it is a bit ugly in a type signature IMO.